pkg/webui/components/key-value-map/entry.js

```diff
@@ -84,12 +89,16 @@ class Entry extends React.Component {
 
 Entry.propTypes = {
   index: PropTypes.number.isRequired,
+  indexAsKey: PropTypes.bool.isRequired,
   keyPlaceholder: PropTypes.message.isRequired,
   name: PropTypes.string.isRequired,
   onBlur: PropTypes.func.isRequired,
   onChange: PropTypes.func.isRequired,
   onRemoveButtonClick: PropTypes.func.isRequired,
-  value: PropTypes.shape({ key: PropTypes.string, value: PropTypes.string }).isRequired,
+  value: PropTypes.shape({
+    key: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    value: PropTypes.any,
+  }).isRequired,
```

Comment:
```suggestion
  }),
```

In your implementation, the value can be `undefined`.